.github/workflows/check.yml

```diff
@@ -0,0 +1,161 @@
+name: check
+on:
+  push:
+  pull_request:
+  schedule:
+    - cron: "0 8 * * *"
+
+concurrency:
+  group: check-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  test:
+    name: test ${{ matrix.py }} - ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        py:
+          - "3.10"
+          - "pypy-3.7-v7.3.7"  # ahead to start it earlier because takes longer
+          - "pypy-2.7-v7.3.6"  # ahead to start it earlier because takes longer
+          - "3.9"
+          - "3.8"
+          - "3.7"
+          - "3.6"
+          - "3.5"
+          - "2.7"
+        os:
+          - ubuntu-20.04
+          - windows-2022
+          - macos-10.15
+
+    steps:
+      - name: Setup python for tox
+        uses: actions/setup-python@v2
+        with:
+          python-version: "3.10"
+      - name: Install tox
+        run: python -m pip install tox
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: Setup python for test ${{ matrix.py }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.py }}
+      - name: Pick environment to run
+        run: |
+          import codecs
+          import os
+          import platform
+          import sys
+          cpy = platform.python_implementation() == "CPython"
+          base =("{}{}{}" if cpy else "{}{}").format("py" if cpy else "pypy", *sys.version_info[0:2])
+          env = "TOXENV={}\n".format(base)
+          print("Picked:\n{}for{}".format(env, sys.version))
+          with codecs.open(os.environ["GITHUB_ENV"], "a", "utf-8") as file_handler:
+               file_handler.write(env)
```

Comment:
this can probably just run `tox -e py` maybe?

Reply:
That would not work for two reasons:

- we're currently always using tox with 3.10 to run the test suite of the tox tool, and -e py would mean we'd always test under 3.10.
- we'd have to massage the coverage filenames to differentiate between runs, as uploading the same artefact twice operates in overwrite mode.

Reply:
ahh makes sense